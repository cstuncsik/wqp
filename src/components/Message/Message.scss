.Message-item {
  display: inline-block;
  padding: 20px;
  margin: 20px 0 0;
  color: #ccc;

  &__success {
    background: #080;
  }

  &__error {
    background: #c00;
  }

  button {
    margin-left: 20px;
  }
}
