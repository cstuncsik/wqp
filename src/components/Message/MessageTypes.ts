export interface Message {
  id: string
  type: 'success' | 'info' | 'warning' | 'error'
  text: string
}

export interface MessageItemFCProps {
  message: Message
  onClose: () => void
}
