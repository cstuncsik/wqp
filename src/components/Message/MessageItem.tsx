import React, { FC } from 'react'
import { MessageItemFCProps } from 'components/Message/MessageTypes'

export const MessageItem: FC<MessageItemFCProps> = ({ message, onClose }) => {
  return (
    <div className={`Message-item Message-item__${message.type}`}>
      {message.text}
      <button type="button" onClick={onClose}>
        X
      </button>
    </div>
  )
}
