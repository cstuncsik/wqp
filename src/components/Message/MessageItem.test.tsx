import React from 'react'
import { render } from '@testing-library/react'

import { MessageItem } from 'components/Message/MessageItem'
import { Message } from 'components/Message/MessageTypes'

const messageTypes: Message['type'][] = ['success', 'info', 'warning', 'error']
const testCases: [Message['type'], Message][] = messageTypes.map(type => [
  type,
  {
    id: '1b31592d-3d8a-42ff-9a8a-a373b10c2ade',
    type,
    text: type + ' message text'
  }
])

describe('MessageItem component', () => {
  test.each(testCases)('renders %s message', (type, message) => {
    const { container } = render(<MessageItem message={message} onClose={() => {}} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
