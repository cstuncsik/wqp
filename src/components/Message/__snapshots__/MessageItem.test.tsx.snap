// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem component renders error message 1`] = `
<div
  class="Message-item Message-item__error"
>
  error message text
  <button
    type="button"
  >
    X
  </button>
</div>
`;

exports[`MessageItem component renders info message 1`] = `
<div
  class="Message-item Message-item__info"
>
  info message text
  <button
    type="button"
  >
    X
  </button>
</div>
`;

exports[`MessageItem component renders success message 1`] = `
<div
  class="Message-item Message-item__success"
>
  success message text
  <button
    type="button"
  >
    X
  </button>
</div>
`;

exports[`MessageItem component renders warning message 1`] = `
<div
  class="Message-item Message-item__warning"
>
  warning message text
  <button
    type="button"
  >
    X
  </button>
</div>
`;
